src/components/button/index.jsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable react/button-has-type */
+/*
+	Disabled ESLint rule as the type attribute is using oneOf propType with a default
+	value - which ESLint is unable to parse as acceptable when in the use case below
+	is valid as there is a default and the use of oneOf PropType validation
+*/
+import { forwardRef } from "react";
+import PropTypes from "prop-types";
+import classnames from "classnames";
+
+const COMPONENT_CLASS_NAME = "c-button";
+
+const Button = forwardRef((props, ref) => {
+	const {
+		accessibilityLabel,
+		children,
+		className,
+		disabled,
+		fullWidth,
+		href,
+		iconLeft,
+		iconRight,
+		size,
+		style,
+		type,
+		...rest
+	} = props;
+
+	const defaultProps = {
+		...rest,
+		ref,
+		"aria-label": accessibilityLabel,
+		className: classnames({
+			[`${COMPONENT_CLASS_NAME}`]: true,
+			[`${COMPONENT_CLASS_NAME}--link`]: href,
+			[`${COMPONENT_CLASS_NAME}--full-width`]: fullWidth,
+			[`${COMPONENT_CLASS_NAME}--${size}`]: size,
+			[`${COMPONENT_CLASS_NAME}--${style}`]: style,
+			[`${className}`]: className,
+		}),
+	};
+
+	const buttonContents = (
+		<>
+			{iconLeft}
+			<span>{children}</span>
+			{iconRight}
+		</>
+	);
+
+	if (href) {
+		return (
+			<a {...defaultProps} href={href}>
+				{buttonContents}
+			</a>
+		);
+	}
+	return (
+		<button {...defaultProps} type={type} aria-disabled={disabled}>
```

Comment:
Am I correct that `aria-disabled` allows a screen reader to see the button, but the button isn't disabled? Also, we shouldn't use `disabled` as that makes the button less inclusive. Will we need to write checks for this attribute into event handlers in other components, blocks as well as documenting that requirement?